blocks/api/parser.js

```diff
@@ -181,12 +182,23 @@ export function getAttributesFromDeprecatedVersion( blockType, innerHTML, attrib
 		try {
 			// Handle migration of older attributes into current version if necessary.
 			const deprecatedBlockAttributes = getBlockAttributes( deprecatedBlockType, innerHTML, attributes );
-			const migratedBlockAttributes = deprecatedBlockType.migrate ? deprecatedBlockType.migrate( deprecatedBlockAttributes ) : deprecatedBlockAttributes;
 
 			// Attempt to validate the parsed block. Ignore if the the validation step fails.
 			const isValid = isValidBlock( innerHTML, deprecatedBlockType, deprecatedBlockAttributes );
 			if ( isValid ) {
-				return migratedBlockAttributes;
+				const migratedBlockAttributesAndInnerBlocks = deprecatedBlockType.migrate &&
+					deprecatedBlockType.migrate( deprecatedBlockAttributes, innerBlocks );
+
+				if ( isArray( migratedBlockAttributesAndInnerBlocks ) ) {
```

Comment:
Minor: Can we simplify the logic here or is it too clever?

`const [ migratedAttributes, migratedInnerBlocks = innerBlocks ] = castArray( migratedBlockAttributesAndInnerBlocks );`

Reply:
I like this version castArray simplifies the logic, I updated to use it :)